src/ConfirmationInput/ConfirmationInput.tsx

```diff
@@ -0,0 +1,217 @@
+import * as React from 'react';
+import cx from '../utils/classnames';
+import { props, t } from '../utils';
+import omit = require('lodash/omit');
+import * as InputChildren from 'react-input-children';
+import FlexView from 'react-flexview';
+import Icon from '../Icon/Icon';
+
+export type ConfirmationInputRequiredProps = {
+  /** input placeholder */
+  placeholder?: string,
+  /** labels for 'clear' and 'toConfirm' buttons */
+  text: {
+    clear?: string,
+    toConfirm?: string
+  },
+  /** icons for 'clear' and 'toConfirm' buttons */
+  icon: {
+    clear?: string,
+    toConfirm?: string
+  },
+  /** an optional class name to pass to top level element of the component */
+  className?: string,
+  /** component's id */
+  id?: string,
+  /** an optional style object to pass to top level element of the component */
+  style?: React.CSSProperties
+}
+
+export type ConfirmationInputDefaultProps = {
+  /** initial value */
+  initialValue: string,
+  /** true if disabled */
+  disabled?: boolean,
+  /** called when input box content is changed */
+  onChange: (value: string) => void,
+  /** called when confirming input content */
+  onConfirm: (value: string) => void,
+  /** called when clearing confirmed content */
+  onClear: () => void,
+};
+
+export type ConfirmationInputProps = ConfirmationInputRequiredProps & Partial<ConfirmationInputDefaultProps>
+
+export const Props = {
+  initialValue: t.maybe(t.String),
+  onChange: t.maybe(t.Function),
+  onConfirm: t.maybe(t.Function),
+  onClear: t.maybe(t.Function),
+  placeholder: t.maybe(t.String),
+  disabled: t.maybe(t.Boolean),
+  text: t.struct({
+    clear: t.maybe(t.String),
+    toConfirm: t.maybe(t.String)
+  }),
+  icon: t.struct({
+    clear: t.maybe(t.String),
+    toConfirm: t.maybe(t.String)
+  }),
+  className: t.maybe(t.String),
+  id: t.maybe(t.String),
+  style: t.maybe(t.Object)
+};
+
+const defaultProps: ConfirmationInputDefaultProps = {
+  initialValue: '',
+  disabled: false,
+  onChange: () => {},
+  onConfirm: () => {},
+  onClear: () => {}
+};
+
+export type ConfirmationInputState = {
+  focused: boolean,
+  hoveringConfirm: boolean,
+  value: string
+}
+
+export type ConfirmProps = {
+  text?: string,
+  icon?: string,
+  onMouseDown: () => void,
+  onMouseEnter: () => void,
+  onMouseLeave: () => void
+} | undefined
+
+@props(Props, { strict: false })
+export default class ConfirmationInput extends React.PureComponent<ConfirmationInputProps, ConfirmationInputState> {
+
```

Comment:
@veej we should temporarily re-add the `defaultProps` here to avoid breaking `@props` validation.

(like @gabro did in this commit: https://github.com/buildo/react-components/pull/1077/commits/f6dd3aa00f448bc7138507624242d95f72fdeb38)

Reply:
nevermind, we'll do it in a separate issue